### DISCUSSION TITLE
MVTLayer handling 404 error on data loading

### DISCUSSION THREAD

**Discussion by DAN-NGIS** [visgl/deck.gl#6900] [Q&A]
### Description

I am using MVTLayer to render vector tile layers on top of the Google map.
I have generated those tiles out of a GeoJson file that I had and the script I used to generate the tiles didn't generate tiles for locations like oceans where there was no feature existing. for example on zoom 6 and x 49 y 38 I have no pbf in my data set to serve the MVTLayer. so in my browser, I will get a 404 error as that tile is missing.
wondering if there is any way I could catch the response status of the data on the deck gl layer and I handle the 404 by passing an empty result instead of showing a console error?

![image](https://user-images.githubusercontent.com/70558260/167981541-51398b16-230e-49f7-abfa-0f889c6b1e79.png)

response from google cloud storage where I am serving my tiles:
![image](https://user-images.githubusercontent.com/70558260/167981715-8d621029-eb34-42f5-9aa1-7d43538389ca.png)

### Expected Behavior

MVTLayer handles missing tile or requests where there is no tile exists.

### Sample code

    return new MVTLayer({
      id: dataLevel,
      data: [`https://storage.googleapis.com/BUCKET_NAME/province/{z}/{x}/{y}.pbf`],
      visible: layer.settings?.visible,
      opacity:  1,
      pickable: true,
      maxZoom: 10, // we don't have tiles for zoom more than 10 on districts which will be 10 meter resolution
      minZoom: 0,
      getLineColor: [136, 143, 136, 255],
      getFillColor: [255, 255, 255, 0],
      lineWidthMinPixels: 2,
      filled: true,
    });
  }

### Environment

- Framework version: Angular v13, Deck.gl v8.7.8
- Browser: chrome
- OS: Windows

**Comment by Pessimistress** [comment]
To expand on [this answer](https://github.com/visgl/deck.gl/issues/6899#issuecomment-1124640272):

1. You are not specifying the [extent](https://deck.gl/docs/api-reference/geo-layers/tile-layer#extent) prop, which culls all fetch attempts outside of a given bounding box.

2. For missing tiles inside the bounding box, if you have control over the server, requests should return 204 No Content, not 404.

3. If you don't have control over the server, you can implement your own `props.fetch`. See this thread: https://github.com/visgl/deck.gl/discussions/6859#discussioncomment-2745473
